File: Java/src/main/java/QuestionBank/Solution401_500/Solution455.java
package QuestionBank.Solution401_500;

import java.util.Arrays;

/**
 * @author windows
 */
public class Solution455 {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int count = 0;
        for (int i = g.length - 1; i >= 0; i--) {
            for (int j = s.length - 1; j >= 0; j--) {
                if (s[j] >= g[i]){
                    count++;
                    s[j] = -1;
                    break;
                }
            }
        }
        return count;
    }


    public static void main(String[] args) {
        Solution455 solution455 = new Solution455();
    }
}

File: C++/src/Solution401_500/Solution455.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution455.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;
#include <algorithm>

int Solution455::findContentChildren(vector<int> &g, vector<int> &s) {
    sort(g.begin(), g.end());
    sort(s.begin(), s.end());
    int count = 0;
    for (int num : g) {
        for (int i = 0; i < s.size(); ++i) {
            if (s[i] >= num){
                count++;
                s[i] = 0;
                break;
            }
        }
    }
    return count;
}
